### PR TITLE
Step4 posts

### PR DESCRIPTION
**Pull request by jdstriker731**
The development site for this stage of Week 3's assignment:
https://dev-dot-johndallard-step-2020.uc.r.appspot.com

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -25,41 +26,61 @@
...
+   * Converts the Java array into a JSON string using Gson
+   */
+  private String convertToJsonUsingGson(ArrayList<String> comments) {
+    return new Gson().toJson(comments);
+  }
+
   /**
    * Converts the messages ArrayList into a JSON string using manual String concatentation.
    */
   private String convertToJson(ArrayList<String> array) {
```
[the remark this answers is not in the document]
Comment (jdstriker731) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:64]: I don't necessarily need convertToJson() at this point. Going forward I just use the function above it. Should I just remove this function?

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -76,8 +77,43 @@ const fetchMessageUsingJSON = () => {
...
+  fetch('/data').then(response => response.json()).then(comments => {
+    // messagesObj is an object, not a string, so we have to
+    // reference its fields to create HTML content
+    
+    const commentsSize = comments.length;
+
+    // If there are no comments
+    if (commentsSize === 0)
+    {
+      return;
```
Comment (askewc) [inline portfolio/src/main/webapp/script.js:96]: if-returns don't need an else clause. If we didn't return, we're already in the 'else' case

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -76,8 +77,43 @@ const fetchMessageUsingJSON = () => {
...
     factContainer.innerText = message;
   });
 };
+
+const showUserComments = () => {
+  fetch('/data').then(response => response.json()).then(comments => {
+    // messagesObj is an object, not a string, so we have to
+    // reference its fields to create HTML content
+    
+    const commentsSize = comments.length;
```
Comment (askewc) [inline portfolio/src/main/webapp/script.js:91]: no need for extra const here?

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -76,8 +77,43 @@ const fetchMessageUsingJSON = () => {
...
+
+    // If there are no comments
+    if (commentsSize === 0)
+    {
+      return;
+    } else {
+      // Build the comments setion with all of the user comments, one after the other
+      for (let i = 0; i < commentsSize; i++)
```
Comment (askewc) [inline portfolio/src/main/webapp/script.js:99]: move the opening curly braces up:

```
for (...) {
  ...
}

if (...) {
  ...
}
```

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -32,9 +33,9 @@ const randomFactGenerator = () => {
 };
 
 /**
- * Fetches content from the server and adds it to the DOM.
+ * Fetches a the content from the server and adds it to the DOM.
```
Comment (askewc) [inline portfolio/src/main/webapp/script.js:36]: a the

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (askewc) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:64]: yup!

**Pull request merged**